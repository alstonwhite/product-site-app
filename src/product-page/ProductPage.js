import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  useHistory,
  useLocation,
  useParams
} from "react-router-dom";

import Header from '../app-level/Header'
import NavBar from '../app-level/NavBar'
import ProductDetail from './ProductDetail';


const ProductPage = ({products, cart, onAdd}) => {
  
  let params = useParams();
  let product = products.find(p => p.id === parseInt(params.id));
  
  return (
    <div className="product-page" id="product-page">
      <Header/>
      <ProductDetail 
        product={product}
        onAdd={onAdd}
      />
      <NavBar
        cart={cart}
      />
    </div>
  );
};
  
  export default ProductPage;