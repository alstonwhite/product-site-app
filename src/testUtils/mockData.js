export const testSections = 
    [
        {
            id: "0",
            slug: "section1",
            sectionTitle: "Section 1",
            active: true
        },
        {
            id: "1",
            slug: "section2",
            sectionTitle: "Section 2",
            active: true
        }
    ]

export const testProducts = 
    [
        {
            id: "0",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 1",
            description: "Description",
            productSection: {fields: {slug: "section1"}},
            active: true
        },
        {
            id: "1",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 2",
            description: "Description",
            productSection: {fields: {slug: "section1"}},
            active: true
        },
        {
            id: "2",
            image1: {fields: {file: {url: "1"}}},
            image2: {fields: {file: {url: "2"}}},
            image3: {fields: {file: {url: "3"}}},
            price: 5,
            title: "Product 3",
            description: "Description",
            productSection: {fields: {slug: "section2"}},
            active: true
        }
    ]

export const testProduct = 
    {
        id: 0,
        image1: {fields: {file: {url: "1"}}},
        image2: {fields: {file: {url: "2"}}},
        image3: {fields: {file: {url: "3"}}},
        price: 5,
        title: "Product 1",
        description: "Description",
        productSection: {fields: {slug: "section1"}},
        active: true
    }

export const testCartProduct = 
[{
    id: 0,
    image1: {fields: {file: {url: "1"}}},
    image2: {fields: {file: {url: "2"}}},
    image3: {fields: {file: {url: "3"}}},
    price: 5,
    title: "Product 1",
    description: "Description",
    productSection: {fields: {slug: "section1"}},
    active: true,
    quantity: 1
}]
