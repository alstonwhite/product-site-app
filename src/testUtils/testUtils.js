import React from "react";
import { Router, Route } from "react-router-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";

export const renderWithRouter = (ui, 
    {
        path="/",
        route="/",
        history = createMemoryHistory({ initialEntries: [route] }) 
    }={}) => ( 
    render( 
        <Router history={history}> 
            <Route path={path} children={ui} /> 
        </Router> 
    ) 
)