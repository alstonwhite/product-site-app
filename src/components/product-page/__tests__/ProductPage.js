import React from 'react'
import { cleanup, screen } from '@testing-library/react'
import ProductPage from '../ProductPage'
import { testSections, testProducts } from '../../../testUtils/mockData'
import { renderWithRouter } from '../../../testUtils/testUtils'

afterEach(cleanup)

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({
    id: "0",
    }), 
}));

jest.mock('react-redux', () => {
    return {
    ...jest.requireActual('react-redux'), 
    useSelector: () => [],
    useDispatch: jest.fn()
    }
});

test('renders proper product', () => {
    renderWithRouter(<ProductPage products={testProducts} sections={testSections}/>)
    const product = testProducts.find(p => p.id === "0")

    expect(screen.getByText(product.title)).toBeInTheDocument()
})

test('snapshot', () => {
    const {container} = renderWithRouter(<ProductPage products={testProducts} sections={testSections}/>)
    expect(container.firstChild).toMatchSnapshot()
})
