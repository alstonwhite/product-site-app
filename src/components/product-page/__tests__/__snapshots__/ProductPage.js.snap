// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="product-page"
  id="product-page"
>
  <div
    class="nav"
  >
    <button
      aria-controls="nav-list"
      aria-expanded="false"
      class="nav-hamburger btn"
    >
      <span
        class="open"
      >
        â˜° Menu
      </span>
      <span
        class="close"
      >
        x Menu
      </span>
    </button>
    <a
      class="nav__cart-content btn"
      href="/cart"
    >
      <div
        class="nav__cart-items"
      >
        Cart: 0 items
      </div>
      <div
        class="nav__cart-value"
      >
         $0
      </div>
    </a>
    <ul>
      <li>
        <a
          class="nav__item"
          href="/"
        >
          All Products
        </a>
      </li>
      <li
        data-testid="nav-section"
      >
        <a
          class="nav__item"
          href="/category/section1"
        >
          Section 1
        </a>
      </li>
      <li
        data-testid="nav-section"
      >
        <a
          class="nav__item"
          href="/category/section2"
        >
          Section 2
        </a>
      </li>
    </ul>
  </div>
  <div
    class="product-detail"
    id="product-detail"
  >
    <div
      class="product-detail__img-gallery"
    >
      <img
        alt="logo"
        class="product-detail__img-main"
        data-testid="imgMain"
        src="1"
      />
      <img
        alt="logo"
        class="product-detail__img-thumb1"
        src="1"
      />
      <img
        alt="logo"
        class="product-detail__img-thumb2"
        data-testid="img2"
        src="2"
      />
      <img
        alt="logo"
        class="product-detail__img-thumb3"
        src="3"
      />
    </div>
    <div
      class="product-detail__content"
    >
      <div
        class="product-detail__content__name"
      >
        Product 1
      </div>
      <div
        class="product-detail__content__price"
      >
        <span>
          $
        </span>
        5
      </div>
      <div
        class="product-detail__content__description"
      >
        Description
      </div>
      <button
        class="product-detail__content__add-cart btn"
      >
        Add to Cart
      </button>
    </div>
  </div>
</div>
`;
